doc/cephadm/host-management.rst

```diff
@@ -48,7 +48,12 @@ To add each new host to the cluster, perform two steps:
 Removing Hosts
 ==============
 
-If the node that want you to remove is running OSDs, make sure you remove the OSDs from the node.
+If the node that want you to remove is running OSDs, make sure you remove the
+OSDs from the node.
+By default cephadm will not allow to remove a host with ceph daemons running in
+it.
+The ``host rm`` command has a ``force`` parameter in order to allow the removal
+operation, although is not recommended to use it.
```

Comment:
We need a solution for node-exporter and crash. If you call `cpeh orch daemon rm crash.host35`, cephadm will instantly deploy a new crash daemon on host 35, which makes it super hard to remove any host.

Reply:
@sebastian-philipp : I think that we must separate things. 
The goal of this PR is only to avoid to remove hosts with daemons running. So unless you use the "force" parameter nobody is going to be able to go into a weird situation. Is just a safe measure until we will have a command like:

"ceph orch host drain"

This command must take care to remove all the daemons in the host. I think that the way to do that is to introduce a new state of the host "draining" .. and let the scheduler to do the job. we can discuss about how to do it.

Meanwhile, and as it is stated in the documentation, the user must take care of removing the daemons running in the hosts changing manually the placement of every service where the host is included.

Reply:
> Meanwhile, and as it is stated in the documentation, the user must take care of removing the daemons running in the hosts changing manually the placement of every service where the host is included.

Imagine you have a thousand hosts or so. And you want to remove a single host. And node-exporter is deployed on all hosts. You need to somehow prevent cephadm from deploying a new node-exporter on that host. Otherwise using `--force` is the **only** way to remove a host from the cluster making this flag pretty useless.

Reply:
@sebastian-philipp :  
This pr does not try to avoid deploy daemons in hosts ... the only target of  this pr is just to avoid to remove a host with daemons running without a explicit warn about that.
So what  I  am doing is just to provide this warn ... but keep the functionality intact using --force (remove the host) if the user really wants to remove the host even with daemons running.

As it is explained in the documentation,, the user must remove manually all the daemons in the node before to proceed to remove the host. (and yes ... we have a problem because the user is not going to be able to remove the node exporter)
To avoid to deploy node-exporter in a host that the user has deleted, we need to implement a "host drain" feature, and probably we will decide to call it  from "rm host" if the user wants to force the operation.
And I would prefer to implement "host drain" in another different pr.

Reply:
I'm somewhat confused .. this PR implies usage of the `--force` flag when I think we are actually trying to discourage it in the doc?

what if we add a more clear workflow on the recommended steps :

1. put the node into maintence mode (`ceph orch host maintenance enter...`)
2. manually remove each daemon (`ceph orch daemon rm ...`)
3. remove the host (`ceph orch host rm ...`)
...?

Reply:
@mgfritch : 
Is very humble the target of this pr.
- without this change: The user can delete a host with daemons running without any warning or any impediment
- with this change: When the user tries to delete a host, the user is warned if there are daemons running in the host and stops the operation. The peration is still possible using the force parameter ( not recommended)

Next an more ambitious steps:
- Implement a drain method, that will remove automagically all the daemons running in a host
- call this method from the "host rm" rm command.

Reply:
I think we're talking past each other here. It's not that this PR is incomplete or wrong, but we have to provide some form of guidance for users.  We have to provide a documentation that actually works and provides a reasonable workflow. We have to change the documentation to a point where users can follow it.

Reply:
I think that it is explained in the current "Removing Hosts" point in the documentation. But we can extend the explanation or make it more clear. Just propose the things you want to clarify or explain better.

Reply:
maybe extend this to be a list of steps (similar to the format of the `Add Host` section)? But specifically describing how to complete the entire process without using the `--force` flag?